components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/db/PinnedSiteEntity.kt

```diff
@@ -28,4 +31,25 @@ internal data class TopSiteEntity(
 
     @ColumnInfo(name = "created_at")
     var createdAt: Long = System.currentTimeMillis()
-)
+) {
+    internal fun toTopSite(): TopSite {
+        val type = if (isDefault) DEFAULT else PINNED
+        return TopSite(
+            id,
+            title,
+            url,
+            createdAt,
+            type
+        )
+    }
+}
+
+internal fun TopSite.toPinnedSite(): PinnedSiteEntity {
+    return PinnedSiteEntity(
+        id = id,
+        title = title ?: "",
+        url = url,
+        isDefault = type === DEFAULT,
+        createdAt = createdAt!!
```

Comment:
```suggestion
        createdAt = createdAt ?: 0
```
nit: avoid double-bangs where possible. This can cause crashes if the extension function is used on a non-pinned top site in error.